### ISSUE TITLE
When making a call, if we get 401, we should handle it same way as 407 and provide credentials

### ISSUE THREAD

**Issue by ocarriles** [RestComm/webrtcomm#90]
[Registration](https://github.com/RestComm/restcomm-web-sdk/blob/master/samples/hello-world/scripts/WebRTComm.js#L1757) deals with 401/407 responses in order to re-authenticate

[Invites](https://github.com/RestComm/restcomm-web-sdk/blob/master/samples/hello-world/scripts/WebRTComm.js#L874) just deal with 407.

My use case is returning 401 Unauthorized

**Comment by atsakiridis** [comment]
@ocarriles can you elaborate more on your scenario? You are using webrtcomm sdk to make a call and the server returns 401? What happens then, the call fails?

**Comment by ocarriles** [comment]
@atsakiridis 
Call fails.
Adding also 401 return status to INVITE processing solves my issue,  just like it is done in the REGISTER code space.

**Comment by atsakiridis** [comment]
@ocarriles, great, so you patched webrtcomm to handle 401 like a 407 for the INVITE, right? Would you like to contribute that back to the codebase? Seems the right thing to do

**Comment by ocarriles** [comment]
@atsakiridis , yep,  but as I do not want to pollute your code, I am wondering if the this use-case is RFC compliant, Perharphs the SBC has to deal with 401 and convert to 407 in same cases. :)

**Comment by atsakiridis** [comment]
I see your point actually it's a little foggy on how the client should handle the following responses from server which seem relevant:

- 401 Unauthorized
- 403 Forbidden
- 407 Proxy Authentication Required

So far I think we 're mostly answering with auth information to 407. In my mind 403 is more like an auth error that doesn't solicit further client requests. But I think we could answer to 401 as well in the same manner

@deruelle what do you think?

**Comment by deruelle** [comment]
@atsakiridis right. Let's handle it for 401 in a similar way than for 407. Please note 401 and 407 doesn't use the same header name (Authentication vs Proxy-Authenticate SIP Header IIRC)

**Comment by atsakiridis** [comment]
Great, in that case, @ocarriles can you create a Pull Request with your changes, so that we can review and move it forward?

Also, from wikipedia the header names are as I pasted above: `401 Unauthorized` and `407 Proxy Authentication Required`.

**Comment by atsakiridis** [comment]
Hello @ocarriles do you think you 'll be able to create the Pull Request with your solution?

**Comment by ocarriles** [comment]
Hi @atsakiridis 
find patched on [WebRTComm](https://github.com/RestComm/sbc/blob/master/application/src/main/webapp/webrtc/phone/scripts/WebRTComm.js)

need to remove:

https://github.com/RestComm/sbc/blob/master/application/src/main/webapp/webrtc/phone/scripts/WebRTComm.js#L14

https://github.com/RestComm/sbc/blob/master/application/src/main/webapp/webrtc/phone/scripts/WebRTComm.js#L3506
https://github.com/RestComm/sbc/blob/master/application/src/main/webapp/webrtc/phone/scripts/WebRTComm.js#L3507

and the whole method "forcePublicCandidate"

**Comment by atsakiridis** [comment]
@ocarriles I'm lost a bit. Why are you editing [sbc](https://github.com/RestComm/sbc) code and not [webrtcomm](https://github.com/RestComm/webrtcomm) code? Aren't the changes you suggest webrtcomm ones? 

Also you can do a [Pull Request](https://help.github.com/articles/about-pull-requests/) that is easier both for you to generate and for us to merge. Do you think you can do that?

**Comment by ocarriles** [comment]
@atsakiridis 
I am the [SBC](https://github.com/RestComm/sbc) author.
As the RestComm/sbc injects a proof of concept webrtc client dependecy, I decided to bundle my own webrtcomm code.
Hope it helps.

**Comment by atsakiridis** [comment]
I see. Thanks @ocarriles will try to fix the webrtcomm code so that when inviting if we get 401, we handle it same way as 407.

Let me also update the title to make a little bit more explanatory
